Jenkinsfile_promote_salt_packages

```diff
@@ -9,49 +9,83 @@ properties([
+                    dir('/tmp/salt-promote-pipeline-env') {
+                        if (salt_mu_branch != 0 || salt_packaging_mu_branch != 0) {
+                            echo "MU branches do not exist. Creating them"
+                            sh "git clone --branch openSUSE/release/${salt_version} --depth 1 https://github.com/openSUSE/salt"
+                            dir('/tmp/salt-promote-pipeline-env/salt') {
+                                sh "git checkout -b openSUSE/MU/${mu_version}"
+                                sh "git push origin openSUSE/MU/${mu_version}"
+                            }
+                            echo "Successfully created and pushed 'openSUSE/MU/${mu_version}' branch to 'openSUSE/salt' repository"
+                            sh "git clone --branch release/${salt_version} --depth 1 https://github.com/openSUSE/salt-packaging"
+                            dir('/tmp/salt-promote-pipeline-env/salt-packaging') {
+                                sh "git checkout -b MU/${mu_version}"
+                                sh "git push origin MU/${mu_version}"
+                            }
+                            echo "Successfully created and pushed 'MU/${mu_version}' branch to 'openSUSE/salt-packaging' repository"
+                        }
+                        else {
+                            if (params.recreate_salt_mu_branches) {
+                                // Forcing recreation of MU branches
+                                echo "The MU branches already exist, but pipeline was set to recreate them"
+                                sh "git clone --branch openSUSE/release/${salt_version} --depth 1 https://github.com/openSUSE/salt"
+                                dir('/tmp/salt-promote-pipeline-env/salt') {
+                                    sh "git checkout -b openSUSE/MU/${mu_version}"
+                                    sh "git push -f origin openSUSE/MU/${mu_version}"
+                                }
+                                echo "Successfully recreated and pushed 'openSUSE/MU/${mu_version}' branch to 'openSUSE/salt' repository"
+                                sh "git clone --branch release/${salt_version} --depth 1 https://github.com/openSUSE/salt-packaging"
+                                dir('/tmp/salt-promote-pipeline-env/salt-packaging') {
+                                    sh "git checkout -b MU/${mu_version}"
+                                    sh "git push -f origin MU/${mu_version}"
+                                }
+                                echo "Successfully recreated and pushed 'MU/${mu_version}' branch to 'openSUSE/salt-packaging' repository"
+                            }
+                            else {
+                                error("The MU branches were already created. Default promotion is not possible. Exiting.")
+                            }
+                        }
+                        sh "rm salt -rf && rm salt-packaging -rf"
```

Comment:
In case of `error()`, that particular call to `rm` won't run but then, the general post execution cleanup will take care of removing the whole temporary environment (including those directories):

https://github.com/openSUSE/salt-package-promote-obs/blob/9700d9c981399bd4ca9880e8a1ba53ac1765c619/Jenkinsfile_promote_salt_packages#L125-L129

So, this shouldn't be an issue.